var/spack/repos/builtin/packages/pythia8/package.py

```diff
@@ -132,17 +132,21 @@ class Pythia8(AutotoolsPackage):
     filter_compiler_wrappers("Makefile.inc", relative_root="share/Pythia8/examples")
 
     @run_before("configure")
-    def setup_cxxstd(self):
+    def setup_configure(self):
         filter_file(
             r"-std=c\+\+[0-9][0-9]", f"-std=c++{self.spec.variants['cxxstd'].value}", "configure"
         )
 
-    # Fix for https://gitlab.com/Pythia8/releases/-/issues/428
-    @when("@:8.311")
-    def patch(self):
-        filter_file(
-            r"[/]examples[/]Makefile[.]inc\|;n' \\", "/examples/Makefile.inc|' \\", "configure"
-        )
+        # Fix for https://gitlab.com/Pythia8/releases/-/issues/428
+        with when("@:8.311"):
+            filter_file(
+                r"[/]examples[/]Makefile[.]inc\|;n' \\", "/examples/Makefile.inc|' \\", "configure"
+            )
+
+    # Fix for https://gitlab.com/Pythia8/releases/-/issues/523
+    with when("@:8.312 cxxstd=20"):
+        patch("pythia8-cpp20-fjcore-forward-decl.patch", when="%apple-clang")
```

Comment:
I'd suggest applying the patch regardless of compiler. There are other llvm-based compilers that may need this same patch. The code is correct with the patch. When a future new version is added by someone using gcc, their build tests will verify that the patch still applies cleanly.

Reply:
Fair enough

Reply:
@wdconinc I've removed the compiler constraint now.
I also added a lower version bound so that this patch applies to `@8.307:8.312` and I've verified that the patch applies and compiler with at least these versions.